pkg/controller/certificates/sync.go

```diff
@@ -218,6 +224,16 @@ func (c *Controller) updateSecret(name, namespace string, cert, key []byte) (*ap
 	}
 	secret.Data[api.TLSCertKey] = cert
 	secret.Data[api.TLSPrivateKeyKey] = key
+
+	if secret.Annotations == nil {
+		secret.Annotations = make(map[string]string)
+	}
+
+	secret.Annotations[altNamesAnnotation] = strings.Join(spec.DNSNames, ",")
+	secret.Annotations[commonNameAnnotation] = spec.CommonName
```

Comment:
We shouldn't directly copy these values from the Certificate resource - they default in peculiar ways (see here: https://github.com/kragniz/cert-manager/blob/f6210c12c627155a580ce21f041b81e610e3051a/pkg/controller/certificates/sync.go#L100-L107)